lib/datastore/request.js

```diff
@@ -199,6 +199,30 @@ DatastoreRequest.prototype.insert = function(entities, callback) {
  * }, function(err) {});
  *
  * //-
+ * // Save different types of data, including ints, doubles, dates, booleans,
+ * // blobs, and lists.
+ * //
+ * // Notice that we are providing an incomplete key. After saving, the original
+ * // Key object used to save will be updated to contain the path with its
+ * // generated ID.
+ * //-
+ * var key = dataset.key('Company');
+ *
+ * dataset.save({
+ *   key: key,
+ *   data: {
+ *     name: 'DonutShack', // strings
+ *     rating: datastore.int(8), // ints
+ *     worth: datastore.double(123456.78), // doubles
+ *     numDonutsServed: 45, // detect number type (int or double)
+ *     founded: new Date('Tue May 12 2015 15:30:00 GMT-0400 (EDT)'), // dates
+ *     isStartup: true, // booleans
+ *     donutEmoji: new Buffer('\uD83C\uDF69'), // buffers
+ *     keywords: ['donut', 'coffee', 'yum'] // lists of objects
+ *   }
+ * }, function(err) {});
```

Comment:
Could we add a line here that shows how to mess with the new key? like `console.log(key.<something>)` ?